File: src/echo-microservices/app-catalog/src/main/java/echo/microservice/catalog/repository/ChapterRepository.java
package echo.microservice.catalog.repository;

import echo.core.domain.entity.ChapterEntity;
import echo.core.domain.entity.EpisodeEntity;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;

import java.util.List;

/**
 * @author Maximilian Irro
 */
public interface ChapterRepository extends JpaRepository<ChapterEntity,Long> {

    List<ChapterEntity> findAllByEpisode(EpisodeEntity episode);

    @Query("SELECT DISTINCT chapter FROM ChapterEntity chapter WHERE chapter.episode.exo = :exo")
    List<ChapterEntity> findAllByEpisodeExo(@Param("exo") String exo);

}

File: src/echo-actors/src/main/scala/echo/actor/catalog/repository/ChapterRepository.scala
package echo.actor.catalog.repository

import echo.core.domain.entity.{ChapterEntity, EpisodeEntity}
import org.springframework.data.jpa.repository.{JpaRepository, Query}
import org.springframework.data.repository.query.Param

/**
  * @author Maximilian Irro
  */
trait ChapterRepository extends JpaRepository[ChapterEntity, java.lang.Long] {

    def findAllByEpisode(episode: EpisodeEntity): java.util.List[ChapterEntity]

    @Query("SELECT DISTINCT chapter FROM ChapterEntity chapter WHERE chapter.episode.exo = :episodeExo")
    def findAllByEpisodeExo(@Param("episodeExo") episodeExo: String): java.util.List[ChapterEntity]

}
